shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Generator/Presets.hs

```diff
@@ -87,20 +89,20 @@ someKeyPairs c lower upper =
     <$> QC.choose (lower, upper)
     <*> QC.shuffle (keyPairs c)
 
-mSigBaseScripts :: Era era => Constants -> MultiSigPairs era
-mSigBaseScripts c = mkMSigScripts (keyPairs c)
+mSigBaseScripts :: forall era. EraGen era => Proxy era -> Constants -> ScriptPairs era
+mSigBaseScripts _proxy c = mkMSigScripts @era (keyPairs c)
```

Comment:
You don't need the proxy here, you can just use the type application

Reply:
Actually you can't just use type application here. Note that ScriptPairs is defined
type ScriptPairs era = [(Core.Script era, Core.Script era)]
So the typing of the function is really
mSigCombinedScripts :: forall era. EraGen era => Proxy era -> Constants -> [(Core.Script era, Core.Script era)]

And since Core.Script is a non-injective type family, without the Proxy era, the function will always be ambiguous.